### PR TITLE
Ensure that jakarta json types can be deserialized in native mode

### LINKED ISSUE #45084: A NoSuchMethodException is raised when a REST resource is called with a single JsonObject argument in native build
### Describe the bug

I have a simple resource that exposes a single POST endpoint which consumes JSON, produces a plaintext and has a single argument of type `JsonObject`. When I build this application natively and call the endpoint with a JSON string it raises a `NoSuchMethodException`.

A JVM build works just fine: The endpoint returns the `message` string.

### Expected behavior

The endpoint should return the `message` string from the JSON request object.

### Actual behavior

 It raises a `NoSuchMethodException`:

```log
2024-12-12 10:30:01,625 ERROR [io.qua.ver.htt.run.QuarkusErrorHandler] (executor-thread-1) HTTP Request to /hello failed, error id: d1eeb235-2f75-472c-9e5c-a56e242e9298-1: java.lang.RuntimeException: java.lang.NoSuchMethodException: org.acme.ExampleResource.post(jakarta.json.JsonObject)
        at org.jboss.resteasy.reactive.server.spi.ResteasyReactiveResourceInfo.getMethod(ResteasyReactiveResourceInfo.java:84)
        at org.jboss.resteasy.reactive.server.spi.ResteasyReactiveResourceInfo.getParameterAnnotations(ResteasyReactiveResourceInfo.java:125)
        at org.jboss.resteasy.reactive.server.handlers.RequestDeserializeHandler.getAnnotations(RequestDeserializeHandler.java:133)
        at org.jboss.resteasy.reactive.server.handlers.RequestDeserializeHandler.isReadable(RequestDeserializeHandler.java:118)
        at org.jboss.resteasy.reactive.server.handlers.RequestDeserializeHandler.handle(RequestDeserializeHandler.java:78)
        at io.quarkus.resteasy.reactive.server.runtime.QuarkusResteasyReactiveRequestContext.invokeHandler(QuarkusResteasyReactiveRequestContext.java:135)
        at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:147)
        at io.quarkus.vertx.core.runtime.VertxCoreRecorder$15.runWith(VertxCoreRecorder.java:637)
        at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2675)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2654)
        at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1627)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1594)
        at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:11)
        at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:11)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base@21/java.lang.Thread.runWith(Thread.java:1596)
        at java.base@21/java.lang.Thread.run(Thread.java:1583)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:833)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.posix.thread.PosixPlatformThreads.pthreadStartRoutine(PosixPlatformThreads.java:211)
Caused by: java.lang.NoSuchMethodException: org.acme.ExampleResource.post(jakarta.json.JsonObject)
        at java.base@21/java.lang.Class.checkMethod(DynamicHub.java:1065)
        at java.base@21/java.lang.Class.getMethod(DynamicHub.java:1050)
        at org.jboss.resteasy.reactive.server.spi.ResteasyReactiveResourceInfo.getMethod(ResteasyReactiveResourceInfo.java:79)
        ... 18 more

```

### How to Reproduce?

1. Create a simple endpoint like this:

```java
@Path("/hello")
public class ExampleResource {

    @POST
    @Consumes(MediaType.APPLICATION_JSON)
    @Produces(MediaType.TEXT_PLAIN)
    public String post(JsonObject request) {
        return request.getString("message");
    }
}
```

2. Create a native build of the application and run it
3. Send a http request like this:

```http
POST http://localhost:8080/hello
Content-Type: application/json

{
  "message": "In a bottle"
}
```

For a reproducer see: https://github.com/nikolassv/quarkus-reproducer-rest-nosuchmethodexception

### Output of `uname -a` or `ver`

Linux 5.15.167.4-microsoft-standard-WSL2 #1 SMP Tue Nov 5 00:21:55 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

openjdk version "21" 2023-09-19 OpenJDK Runtime Environment GraalVM CE 21+35.1 (build 21+35-jvmci-23.1-b15) OpenJDK 64-Bit Server VM GraalVM CE 21+35.1 (build 21+35-jvmci-23.1-b15, mixed mode, sharing)

### Mandrel or GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

3.17.4

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.9 (8e8579a9e76f7d015ee5ec7bfcdc97d260186937)

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by geoand**
- Fixes: #45084

### CODE REVIEW

File: independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/providers/serialisers/jsonp/ServerJsonObjectHandler.java [hunk 1]
```diff
@@ -30,4 +31,15 @@ public void writeResponse(JsonObject o, Type genericType, ServerRequestContext c
...
+    //    public boolean isReadable(Class<?> type, Type genericType, ResteasyReactiveResourceInfo lazyMethod,
+    //            MediaType mediaType) {
+    //        return JsonObject.class.isAssignableFrom(type);
+    //    }
+    //
+    //    @Override
+    //    public JsonObject readFrom(Class<JsonObject> type, Type genericType, MediaType mediaType,
+    //            ServerRequestContext context) throws WebApplicationException, IOException {
+    //        return JsonpUtil.reader(context.getInputStream(), mediaType).readObject();
+    //    }
```
Comment (gsmet) [inline independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/providers/serialisers/jsonp/ServerJsonObjectHandler.java:44]: Why this commented out code?

File: integration-tests/qute/src/test/java/io/quarkus/it/qute/QuteTestCase.java [hunk 2]
```diff
@@ -28,6 +28,18 @@ public void testTemplates() throws InterruptedException {
...
+                .contentType("application/json")
+                .body("""
+                        {
+                          "name": "foo"
+                        }
+                        """)
+                .when().post("/json")
+                .then()
+                .statusCode(200)
+                .body(containsString("foo"));
```
Comment (gsmet) [inline integration-tests/qute/src/test/java/io/quarkus/it/qute/QuteTestCase.java:42]: Could we add the test in something more core than the Qute ITs. Reasoning is that if at some point we drop Qute, we will lose these tests which are related to basic core features.

(Yeah, sorry for being annoying!)

**Review by gsmet — COMMENTED** [review]
I added some questions.

I'm curious though, how can this lead to the method not being registered in native?

File: independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/providers/serialisers/jsonp/ServerJsonObjectHandler.java (hunk 1 above)
Comment (geoand) [reply independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/providers/serialisers/jsonp/ServerJsonObjectHandler.java:44]: By mistake, I'll fix ir

**Comment by geoand** [comment]
> I'm curious though, how can this lead to the method not being registered in native?

In Quarkus REST the built in readers and writers don't need access to the annotations, and using our custom interfaces ensures that

File: integration-tests/qute/src/test/java/io/quarkus/it/qute/QuteTestCase.java (hunk 2 above)
Comment (geoand) [reply integration-tests/qute/src/test/java/io/quarkus/it/qute/QuteTestCase.java:42]: I looked for another place and didn't find a good candidate. 

Agreed it's not a good place, but I have no better candidates in mind

**Review by gastaldi — APPROVED** [review]

**Pull request merged**